messenger/http_transporter.go

```diff
@@ -181,6 +181,14 @@ func (t *HTTPTransporter) Install(msgName string) {
 func (t *HTTPTransporter) Listen() error {
 	host := t.upid.Host
 	port := t.upid.Port
+
+	if host == "" {
+		host = "0.0.0.0"
+	}
+
+	if port == "" {
+		port = "0"
```

Comment:
@fmpwizard So this is not for the slave, but for custom executors. Custom executors need to listen on a port for connecting to the slaves. And port 0 allows the custom executor to listen on any ephemeral port which is free.